Content.Shared/Radio/EntitySystems/SharedHeadsetSystem.cs

```diff
@@ -35,4 +44,45 @@ protected virtual void OnGotUnequipped(EntityUid uid, HeadsetComponent component
     {
         component.IsEquipped = false;
     }
+
+    private void OnGetVerbs(EntityUid uid, HeadsetComponent component, ref GetVerbsEvent<Verb> args)
+    {
+        if (!args.CanInteract || args.Hands == null)
+            return;
+
+        if (component.IsEquipped && Transform(uid).ParentUid != args.User)
+            return;
+
+        if (!_interaction.InRangeUnobstructed(args.User, args.Target))
+            return;
+
+        if (!TryComp(uid, out EncryptionKeyHolderComponent? keyHolder))
+            return;
+
+        foreach ((var channel, var index) in keyHolder.Channels.Select(static (channel, index) => (channel, index)))
+        {
+            var name = _prototype.Index<RadioChannelPrototype>(channel).LocalizedName;
+
+            var toggled = component.ToggledSoundChannels.Contains(channel);
+
+            args.Verbs.Add(new()
+            {
+                Text = toggled ? $"[bold]{name}" : name,
+                Priority = index,
+                Category = VerbCategory.ToggleHeadsetSound,
+                Act = () => ToggleHeadsetSound((uid, component), channel, !toggled)
+            });
+        }
+    }
+
+    /// <summary>
+    /// Toggles channel on given headset to on or off.
+    /// </summary>
+    public static void ToggleHeadsetSound(Entity<HeadsetComponent> headset, string channel, bool on)
```

Comment:
It would also be nice if `on` had a more descriptive comment specifically! Its a little weird but it might just be a me issue.